docs/content/direct/hub-chart.md

```diff
@@ -0,0 +1,263 @@
+
+```shell
+helm upgrade --install ks-hub oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION
+```
+
+The above command will install KubeFlex and the Post Create Hooks, but no Control Planes.
+Please remeber to add `--set "kubeflex-operator.isOpenShift=true"`, when installing into an OpenShift cluster.
+
+User defined control planes can be added using additional value files of `--set` arguments, _e.g._:
+
+- add a single ITS named its1 of default vcluster type: `--set-json='ITSes=[{"name":"its1"}]'`
+- add two ITSes named its1 and its2 of of type vlcuster and host, respectevely: `--set-json='ITSes=[{"name":"its1"},{"name":"its2","type":"host"}]'`
+- add a single WDS named wds1 of default k8s type connected to the one and only ITS: `--set-json='WDSes=[{"name":"wds1"}]'`
+
+A KubeStellar Hub installation compatible with the common setup suitable for Common Setup described in the [examples](./examples.md) could be achieved with the following command:
+
+```shell
+helm upgrade --install ks-hub oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set-json='ITSes=[{"name":"its1"}]' \
+  --set-json='WDSes=[{"name":"wds1"}]'
+```
+
+After the initial instllation is completed, there are two main ways to install additional control planes (_e.g._, create a second `wds2` WDS):
+
+1. Upgrade the initial chart. This choice requires to relist the existing control planes, which would otherwise be deleted:
+
+```shell
+helm upgrade --install ks-hub oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set-json='ITSes=[{"name":"its1"}]' \
+  --set-json='WDSes=[{"name":"wds1"},{"name":"wds2"}]'
+```
+
+2. Install a new chart with a different name. This choice does not requires to relist the existing control planes, but requies to disable the reinstallation of KubeFlex and PCHs:
+
+```shell
+helm upgrade --install add-wds2 oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set='kubeflex-operator.install=false,InstallPCHs=false' \
+  --set-json='WDSes=[{name":"wds2"}]'
+```
+
+## Obtaining the Control Planes kubeconfigs
+
+The following code can be used to wait for the Ready state of all the KubeFlex Control Planes in the hosting cluster:
+
+```shell
+echo "Waiting for all Control Planes to be ready..."
+for cpname in `kubectl get controlplane -o name`; do
+  cpname=${cpname##*/}
+  while [[ $(kubectl get cp $cpname -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do
+    echo "Waiting for Control Plane $cpname..."
+    sleep 5
+  done
+  echo $cpname
+done
+```
+
+The kubeconfig of the KubeFlex control planes can be obtained both with and without the use of `kflex` CLI.
+
+1. using `kflex` CLI
+
+The kubeconfig of a `$cpname` Control Plane can be added as a context of the existing kubeconfig by using the command:
+
+```shell
+kflex ctx $cpname

```

Comment:
If using this command, is the wait above necessary first? Does `kflex ctx $cpname` first wait for the named control plane to be ready? If not, could it be changed to do so? (@pdettori ?).

Reply:
https://github.com/kubestellar/kubeflex/issues/258